src/components/overlay/overlay.stories.ts

```diff
@@ -0,0 +1,352 @@
+import { SbbOverlayElement } from './overlay';
+import readme from './readme.md?raw';
+
+import '../title';
+import '../button';
+import '../link';
+import '../form-field';
+import '../image';
+
+// Story interaction executed after the story renders
+const playStory = async ({ canvasElement }: StoryContext): Promise<void> => {
+  const canvas = within(canvasElement);
+
+  await waitForComponentsReady(() =>
+    canvas.getByTestId('overlay').shadowRoot!.querySelector('.sbb-overlay'),
+  );
+
+  await waitForStablePosition(() => canvas.getByTestId('overlay-trigger'));
+
+  const button = canvas.getByTestId('overlay-trigger');
+  await userEvent.click(button);
+};
+
+const expanded: InputType = {
+  control: {
+    type: 'boolean',
+  },
+};
+
+const backButton: InputType = {
+  control: {
+    type: 'boolean',
+  },
+};
+
+const negative: InputType = {
+  control: {
+    type: 'boolean',
+  },
+};
+
+const accessibilityLabel: InputType = {
+  control: {
+    type: 'text',
+  },
+  table: {
+    category: 'Accessibility',
+  },
+};
+
+const accessibilityCloseLabel: InputType = {
+  control: {
+    type: 'text',
+  },
+  table: {
+    category: 'Accessibility',
+  },
+};
+
+const accessibilityBackLabel: InputType = {
+  control: {
+    type: 'text',
+  },
+  table: {
+    category: 'Accessibility',
+  },
+};
+
+const disableAnimation: InputType = {
+  control: {
+    type: 'boolean',
+  },
+};
+
+const basicArgTypes: ArgTypes = {
+  expanded,
+  'back-button': backButton,
+  accessibilityCloseLabel,
+  accessibilityBackLabel,
+  negative,
+  'accessibility-label': accessibilityLabel,
+  'disable-animation': disableAnimation,
+};
+
+const basicArgs: Args = {
+  expanded: false,
+  'back-button': false,
+  accessibilityCloseLabel: 'Close overlay',
+  accessibilityBackLabel: 'Go back',
+  negative: false,
+  'accessibility-label': undefined,
+  'disable-animation': isChromatic(),
+};
+
+const openOverlay = (_event: PointerEvent, id: string): void => {
+  const overlay = document.getElementById(id) as SbbOverlayElement;
+  overlay.open();
+};
+
+const triggerButton = (overlayId: string, triggerId?: string): TemplateResult => html`
+  <sbb-button
+    data-testid=${triggerId || 'overlay-trigger'}
+    aria-haspopup="dialog"
+    aria-controls=${overlayId}
+    size="m"
+    type="button"
+    @click=${(event: PointerEvent) => openOverlay(event, overlayId)}
+  >
+    Open overlay
+  </sbb-button>
+`;
+
+const codeStyle: Args = {
+  padding: 'var(--sbb-spacing-fixed-1x) var(--sbb-spacing-fixed-2x)',
+  borderRadius: 'var(--sbb-border-radius-4x)',
+  backgroundColor: 'var(--sbb-color-smoke-alpha-20)',
+};
+
+const formDetailsStyle: Args = {
+  marginTop: 'var(--sbb-spacing-fixed-4x)',
+  padding: 'var(--sbb-spacing-fixed-4x)',
+  borderRadius: 'var(--sbb-border-radius-8x)',
+  backgroundColor: 'var(--sbb-color-milk)',
+};
+
+const formStyle: Args = {
+  display: 'flex',
+  flexWrap: 'wrap',
+  alignItems: 'center',
+  gap: 'var(--sbb-spacing-fixed-4x)',
+};
```

Comment:
If these const are all passed in the styleMap function, their type can be changed to `Readonly<StyleInfo>`.